runtime/src/main/cpp/Memory.cpp

```diff
@@ -1817,28 +1834,35 @@ void Kotlin_native_internal_GC_resume(KRef) {
 
 void Kotlin_native_internal_GC_stop(KRef) {
 #if USE_GC
-  if (memoryState->toFree != nullptr) {
+  GC_LOG("Kotlin_native_internal_GC_stop\n")
+  if (memoryState->toRelease != nullptr) {
     GarbageCollect();
-    // TODO: what shall we do with toRelease here? Just leak?
+    konanDestructInstance(memoryState->toRelease);
     konanDestructInstance(memoryState->toFree);
     konanDestructInstance(memoryState->roots);
+    memoryState->toRelease = nullptr;
     memoryState->toFree = nullptr;
     memoryState->roots = nullptr;
+    state->gcSuspendCount = 0;
```

Comment:
What if GC is stopped while suspended?

Reply:
Hard to tell what is right, but moved zeroing above.